### PR TITLE
Supply vendors tweaks and fixes

### PR DESCRIPTION
**Pull request by JeserTheFox**
## About The Pull Request

1. Fixed not being able to restock shotgun ammo boxes in APC and Req vendors.
2. Made possible restocking marine armors in uniform vendors.
3. Minor tweaks of content in Dropship Pilot, Dropship Crew Chief, Vehicle Crewman and Squad Specialist vendors.
4. Vendors intended to be indestructible are actually indestructible now.
5. It is possible to restock "/empty" ammo boxes now. (The handmade from cardboard ones).
6. Req vendor can restock flamer fueltanks now.
7. corresponding list now expanded to include many other types.
8. More different types support restocking
9. Added checks that prevent putting not allowed stuff in flare and MRE supply boxes. (Instead of checking all this when restocking).

## Why It's Good For The Game

Fixes, a bit more loadout options, slight QoL of supply staff.

## Changelog

:cl: Jeser
add: Minor content tweaks in Dropship Pilot, Dropship Crew Chief, Vehicle Crewman and Squad Specialist vendors.
add: APC Supply and Requisitions Ammunition vendors support restocking with handmade ammo boxes. (The ones made out of cardboard). Same requirements as for regular ammo boxes apply.
add: Requisitions Ammunition vendor supports restocking flamer fueltanks now.
add: Uniform vendors now support restocking armor in them.
fix: Fixed not being able to restock shotgun shells boxes in APC Supply and Requisitions Ammunition vendors.
/:cl:

### CODE REVIEW

**Comment by sose96** [comment]
What are the dcc/pilot tweaks?

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 1]
```diff
@@ -307,35 +308,58 @@
...
+		if(item_to_stock.type == R[3] || corrected_path && corrected_path == R[3])
+			//magazines handling
 			if(istype(item_to_stock, /obj/item/ammo_magazine))
+				//flamer fuel tanks handling
 				if(istype(item_to_stock, /obj/item/ammo_magazine/flamer_tank))
 					var/obj/item/ammo_magazine/flamer_tank/FT = item_to_stock
-					if(FT.flamer_chem == initial(FT.flamer_chem))
-						to_chat(user, SPAN_WARNING("\The [FT] contains the wrong fuel."))
+					if(FT.flamer_chem != initial(FT.flamer_chem))
+						to_chat(user, SPAN_WARNING("\The [FT] contains not standard fuel."))
```
Comment (stanalbatross) [inline code/modules/vehicles/interior/interactable/vendors.dm:328]:
```suggestion
						to_chat(user, SPAN_WARNING("\The [FT] contains non-standard fuel."))
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 2]
```diff
@@ -307,35 +308,58 @@
...
-					if(M.current_rounds != M.max_rounds)
-						to_chat(user, SPAN_WARNING("Not all the magazines in \the [A] are full."))
+				//shotgun shells ammo boxes handling
+				if(A.handfuls)
+					var/obj/item/ammo_magazine/AM = locate(/obj/item/ammo_magazine) in item_to_stock.contents
+					if(!AM)
+						to_chat(user, SPAN_WARNING("Something is wrong with \the [A], tell a coder."))
+						return
+					if(AM.current_rounds != AM.max_rounds)
+						to_chat(user, SPAN_WARNING("[A] isn't full. Fill it before you can restock it."))
```
Comment (stanalbatross) [inline code/modules/vehicles/interior/interactable/vendors.dm:345]:
```suggestion
						to_chat(user, SPAN_WARNING("\The [A] isn't full. Fill it before you can restock it."))
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 3]
```diff
@@ -307,35 +308,58 @@
...
 						return
-			if(isgun(item_to_stock))
+				else if(A.contents.len < A.num_of_magazines)
+					to_chat(user, SPAN_WARNING("[A] is not full."))
```
Comment (stanalbatross) [inline code/modules/vehicles/interior/interactable/vendors.dm:348]:
```suggestion
					to_chat(user, SPAN_WARNING("[\The A] isn't full."))
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 4]
```diff
@@ -307,35 +308,58 @@
...
+					return
+				else
+					for(var/obj/item/ammo_magazine/M in A.contents)
+						if(M.current_rounds != M.max_rounds)
+							to_chat(user, SPAN_WARNING("Not all magazines in [A] are full."))
```
Comment (stanalbatross) [inline code/modules/vehicles/interior/interactable/vendors.dm:353]:
```suggestion
							to_chat(user, SPAN_WARNING("Not all magazines in \the [A] are full."))
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 5]
```diff
@@ -307,35 +308,58 @@
...
+							return
+			//loose rounds ammo box handling
+			else if(istype(item_to_stock, /obj/item/ammo_box/rounds))
+				var/obj/item/ammo_box/rounds/A = item_to_stock
+				if(A.bullet_amount < A.max_bullet_amount)
+					to_chat(user, SPAN_WARNING("[A] is not full."))
```
Comment (stanalbatross) [inline code/modules/vehicles/interior/interactable/vendors.dm:359]:
```suggestion
					to_chat(user, SPAN_WARNING("\The [A] isn't full."))
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 6]
```diff
@@ -344,11 +368,63 @@
...
+						return
+				//for the ease of finding enough materials to stack, it will be stored in stacks of 10 sheets just like they come in engie vendor
+				else if(S.amount != 10)
+					to_chat(user, SPAN_WARNING("[S] is being stored in [SPAN_HELPFUL("stacks of 10")] for convenience. [(S.amount < 10) ? "Add to" : "Split"] \the [S] stack to make it a stack of 10 before restocking."))
+					return
+			//M94 flare packs handling
+			else if(istype(item_to_stock, /obj/item/storage/box/m94))
+				var/obj/item/storage/box/m94/flare_pack = item_to_stock
+				if(flare_pack.contents.len < flare_pack.max_storage_space)
+					to_chat(user, SPAN_WARNING("[item_to_stock] is not full."))
```
Comment (stanalbatross) [inline code/modules/vehicles/interior/interactable/vendors.dm:390]:
```suggestion
					to_chat(user, SPAN_WARNING("\The [item_to_stock] is not full."))
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 7]
```diff
@@ -344,11 +368,63 @@
...
+				for(var/obj/item/device/flashlight/flare/F in flare_pack.contents)
+					if(F.fuel < 1)
+						to_chat(user, SPAN_WARNING("Some flares in \the [F] are used."))
+						return
+					if(F.type != flare_type)
+						to_chat(user, SPAN_WARNING("Some flares in \the [F] are not of the correct type."))
+						return
+			//Machete holsters handling
+			else if(istype(item_to_stock, /obj/item/storage/large_holster/machete))
+				var/obj/item/weapon/melee/claymore/mercsword/machete/Mac = locate(/obj/item/weapon/melee/claymore/mercsword/machete) in item_to_stock
```
Comment (stanalbatross) [inline code/modules/vehicles/interior/interactable/vendors.dm:406]:
```suggestion
				var/obj/item/weapon/melee/claymore/mercsword/machete/mac = locate(/obj/item/weapon/melee/claymore/mercsword/machete) in item_to_stock
```
snake_case to be used

File: code/modules/projectiles/ammunition.dm [hunk 8]
```diff
@@ -925,6 +925,28 @@ Turn() or Shift() as there is virtually no overhead. ~N
 /obj/structure/magazine_box/attackby(obj/item/W, mob/living/user)
 	if(!item_box.handfuls)
 		if(istypestrict(W,item_box.magazine_type))
+			if(istype(W, /obj/item/storage/box/m94))
+				var/obj/item/storage/box/m94/flare_pack = W
+				if(flare_pack.contents.len < flare_pack.max_storage_space)
+					to_chat(user, SPAN_WARNING("[W] is not full."))
```
Comment (stanalbatross) [inline code/modules/projectiles/ammunition.dm:931]:
```suggestion
					to_chat(user, SPAN_WARNING("\The [W] is not full."))
```

File: code/game/machinery/vending/vendor_types/requisitions.dm [hunk 9]
```diff
@@ -589,3 +663,36 @@ obj/structure/machinery/cm_vending/sorted/uniform_supply
...
+					item_to_stock.unwield(user)
+				user.temp_drop_inv_item(item_to_stock)
+
+			if(isstorage(item_to_stock.loc)) //inside a storage item
+				var/obj/item/storage/S = item_to_stock.loc
+				S.remove_from_storage(item_to_stock, user.loc)
+
+			qdel(item_to_stock)
+			user.visible_message(SPAN_NOTICE("[user] stocks [src] with \a [R[1]]."),
+			SPAN_NOTICE("You stock [src] with \a [R[1]]."))
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/requisitions.dm:695]:
```suggestion
			user.visible_message(SPAN_NOTICE("[user] stocks \the [src] with \a [R[1]]."),
			SPAN_NOTICE("You stock \the [src] with \a [R[1]]."))
```

File: code/game/machinery/vending/vendor_types/requisitions.dm [hunk 10]
```diff
@@ -315,24 +366,51 @@
...
 					return
-				for(var/obj/item/ammo_magazine/M in A.contents)
-					if(M.current_rounds != M.max_rounds)
-						to_chat(user, SPAN_WARNING("Not all magazines in [A] are full."))
-						return
-			if(istype(item_to_stock, /obj/item/ammo_box/rounds))
+				else
+					for(var/obj/item/ammo_magazine/M in A.contents)
+						if(M.current_rounds != M.max_rounds)
+							to_chat(user, SPAN_WARNING("Not all magazines in [A] are full."))
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/requisitions.dm:410]:
```suggestion
							to_chat(user, SPAN_WARNING("Not all magazines in \the [A] are full."))
```

File: code/game/machinery/vending/vendor_types/requisitions.dm [hunk 11]
```diff
@@ -315,24 +366,51 @@
...
 				var/obj/item/ammo_box/magazine/A = item_to_stock
-				if(A.contents.len < A.num_of_magazines)
+				//shotgun shells ammo boxes handling
+				if(A.handfuls)
+					var/obj/item/ammo_magazine/AM = locate(/obj/item/ammo_magazine) in item_to_stock.contents
+					if(!AM)
+						to_chat(user, SPAN_WARNING("Something is wrong with \the [A], tell a coder."))
+						return
+					if(AM.current_rounds != AM.max_rounds)
+						to_chat(user, SPAN_WARNING("[A] isn't full. Fill it before you can restock it."))
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/requisitions.dm:402]:
```suggestion
						to_chat(user, SPAN_WARNING("\The [A] isn't full. You need to fill it before you can restock it."))
```

File: code/game/machinery/vending/vendor_types/requisitions.dm [hunk 12]
```diff
@@ -315,24 +366,51 @@
...
 	for(R in (listed_products))
-		if(item_to_stock.type == R[3] && !istype(item_to_stock,/obj/item/storage))
+		if(item_to_stock.type == R[3] || corrected_path && corrected_path == R[3])
+			//magazines handling
 			if(istype(item_to_stock, /obj/item/ammo_magazine))
+				//flamer fuel tanks handling
+				if(istype(item_to_stock, /obj/item/ammo_magazine/flamer_tank))
+					var/obj/item/ammo_magazine/flamer_tank/FT = item_to_stock
+					if(FT.flamer_chem != initial(FT.flamer_chem))
+						to_chat(user, SPAN_WARNING("\The [FT] contains not standard fuel."))
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/requisitions.dm:386]:
```suggestion
						to_chat(user, SPAN_WARNING("\The [FT] contains non-standard fuel."))
```

File: code/game/machinery/vending/vendor_types/requisitions.dm [hunk 13]
```diff
@@ -145,6 +166,36 @@
...
+			else if(istype(item_to_stock, /obj/item/stack/folding_barricade))
+				var/obj/item/stack/folding_barricade/B = item_to_stock
+				if(B.amount != 3)
+					to_chat(user, SPAN_WARNING("[B] is being stored in [SPAN_HELPFUL("stacks of 3")] for convenience. Add to \the [B] stack to make it a stack of 3 before restocking."))
+					return
+			//M94 flare packs handling
+			else if(istype(item_to_stock, /obj/item/storage/box/m94))
+				var/obj/item/storage/box/m94/flare_pack = item_to_stock
+				if(flare_pack.contents.len < flare_pack.max_storage_space)
+					to_chat(user, SPAN_WARNING("[item_to_stock] is not full."))
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/requisitions.dm:179]:
```suggestion
					to_chat(user, SPAN_WARNING("\The [item_to_stock] is not full."))
```

File: code/game/machinery/vending/vendor_types/requisitions.dm [hunk 14]
```diff
@@ -145,6 +166,36 @@
 					if((A.flags_attach_features & ATTACH_REMOVABLE) && !(is_type_in_list(A, G.starting_attachment_types))) //There are attachments that are default and others that can't be removed
 						to_chat(user, SPAN_WARNING("[G] has non-standard attachments equipped. Detach them before you can restock it."))
 						return
+			//various stacks handling
+			else if(istype(item_to_stock, /obj/item/stack/folding_barricade))
+				var/obj/item/stack/folding_barricade/B = item_to_stock
+				if(B.amount != 3)
+					to_chat(user, SPAN_WARNING("[B] is being stored in [SPAN_HELPFUL("stacks of 3")] for convenience. Add to \the [B] stack to make it a stack of 3 before restocking."))
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/requisitions.dm:173]:
```suggestion
					to_chat(user, SPAN_WARNING("[B]s are being stored in [SPAN_HELPFUL("stacks of 3")] for convenience. Add to \the [B] stack to make it a stack of 3 before restocking."))
```

File: code/game/machinery/vending/vendor_types/requisitions.dm [hunk 15]
```diff
@@ -133,9 +133,30 @@
...
 	for(R in (listed_products))
-		if(item_to_stock.type == R[3] && !istype(item_to_stock,/obj/item/storage))
+		if(item_to_stock.type == R[3] || corrected_path && corrected_path == R[3])
+			//magazines handling (mainly for subtypes)
+			if(istype(item_to_stock, /obj/item/ammo_magazine))
+				//flamer fuel tanks handling
+				if(istype(item_to_stock, /obj/item/ammo_magazine/flamer_tank))
+					var/obj/item/ammo_magazine/flamer_tank/FT = item_to_stock
+					if(FT.flamer_chem != initial(FT.flamer_chem))
+						to_chat(user, SPAN_WARNING("\The [FT] contains not standard fuel."))
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/requisitions.dm:152]:
```suggestion
						to_chat(user, SPAN_WARNING("\The [FT] contains non-standard fuel."))
```
there MUST be a better way to handle this than repeating the same code three times

possibly a can_restock proc on the item that returns true or false?

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**